src/backup.ts

```diff
@@ -0,0 +1,482 @@
+
+export type CreateBackupCallback = LongRunningCallback<Backup>;
+
+export interface CreateBackupGaxOperation extends GaxOperation {
+  // Overridden with more specific type for CreateBackup operation
+  metadata: Metadata &
+    databaseAdmin.spanner.admin.database.v1.ICreateBackupMetadata;
+}
+
+export type CreateBackupResponse = [
+  Backup,
+  CreateBackupGaxOperation,
+  IOperation
+];
+
+export interface CreateBackupOptions {
+  databasePath: string;
+  expireTime: PreciseDate;
+  gaxOptions?: CallOptions;
+}
+
+/**
+ * IBackup structure with backup state enum translated to string form.
+ */
+type IBackupTranslatedEnum = TranslateEnumKeys<
+  databaseAdmin.spanner.admin.database.v1.IBackup,
+  'state',
+  typeof databaseAdmin.spanner.admin.database.v1.Backup.State
+>;
+
+export type GetMetadataResponse = [IBackupTranslatedEnum];
+type GetMetadataCallback = RequestCallback<IBackupTranslatedEnum>;
+
+type UpdateExpireTimeCallback = RequestCallback<
+  databaseAdmin.spanner.admin.database.v1.IBackup
+>;
+
+type DeleteCallback = RequestCallback<void>;
+
+interface BackupRequest {
+  (
+    config: RequestConfig,
+    callback: ResourceCallback<GaxOperation, IOperation>
+  ): void;
+  <T>(config: RequestConfig, callback: RequestCallback<T>): void;
+}
+/**
+ * The {@link Backup} class represents a Cloud Spanner backup.
+ *
+ * Create a `Backup` object to interact with or create a Cloud Spanner backup.
+ *
+ * @class
+ *
+ * @example
+ * const {Spanner} = require('@google-cloud/spanner');
+ * const spanner = new Spanner();
+ * const instance = spanner.instance('my-instance');
+ * const backup = instance.backup('my-backup');
+ */
+class Backup {
+  id: string;
+  formattedName_: string;
+  instanceFormattedName_: string;
+  request: BackupRequest;
+  constructor(instance: Instance, name: string) {
+    this.request = instance.request;
+    this.instanceFormattedName_ = instance.formattedName_;
+    this.formattedName_ = Backup.formatName_(instance.formattedName_, name);
+    this.id = this.formattedName_.split('/').pop() || '';
+  }
+
+  create(options: CreateBackupOptions): Promise<CreateBackupResponse>;
+  create(options: CreateBackupOptions, callback: CreateBackupCallback): void;
+  /**
+   * @typedef {array} CreateBackupResponse
+   * @property {Backup} 0 The new {@link Backup}.
+   * @property {Operation} 1 An {@link Operation} object that can be used to check
+   *     the status of the request.
+   * @property {object} 2 The full API response.
+   */
+  /**
+   * @callback CreateBackupCallback
+   * @param {?Error} err Request error, if any.
+   * @param {Backup} backup The new {@link Backup}.
+   * @param {Operation} operation An {@link Operation} object that can be used to
+   *     check the status of the request.
+   * @param {object} apiResponse The full API response.
+   */
+  /**
+   * Create a backup.
+   *
+   * @method Backup#create
+   * @param {CreateBackupOptions} options Parameters for creating a backup as
```

Comment:
I believe we need a typedef for CreateBackupOptions so users can see it from the docs site, e.g. https://github.com/prunge-helix/nodejs-spanner/blob/c9955eca54c2c8d64f46d5049822140bb8e68eca/src/instance.ts#L361-L372